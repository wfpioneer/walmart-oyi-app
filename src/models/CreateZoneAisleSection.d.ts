import { CreateAisles } from '../state/reducers/Location';

export interface CreateAisleRequest {
  zoneId: number;
  aisles: CreateAisles[]
}

export interface ZoneAislesSectionResponse {
  aisleId: number;
  aisleName: number;
  createdSectionsResponseDto: CreateAisleSectionsResponse;
  sectionCount: number;
  status: number;
}

export interface CreateZoneAisleSectionResponse {
  aisles: ZoneAislesSectionResponse[];
  status: number;
  zoneId: number;
  zoneName: string;
}

export interface CreateAisleResponse {
  aisleId: number;
  aisleName: number;
  status: number;
  message: string;
  createdSectionsResponseDto: CreateAisleSectionsResponse;
}

interface CreateAisleSectionsResponse {
  aisleId: number;
  status: number;
  message: string;
  createdSections: string[]
}
