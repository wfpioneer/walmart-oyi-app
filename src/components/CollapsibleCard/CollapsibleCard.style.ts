import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const styles = StyleSheet.create({
  menuContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingVertical: 10,
    paddingHorizontal: 10,
    borderBottomWidth: 1,
    backgroundColor: COLOR.HAVELOCK_BLUE
  },
  titleContainer: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    flexDirection: 'row'
  },
  titleText: {
    fontSize: 16
  },
  arrowView: {
    flex: 0.1,
    paddingLeft: 5
  }
});

export default styles;
