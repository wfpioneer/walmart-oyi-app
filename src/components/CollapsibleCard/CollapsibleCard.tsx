import React, { ReactElement, ReactNode, useState } from 'react';
import { 
  Text, TextStyle, TouchableOpacity, View 
} from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import styles from './CollapsibleCard.style';
import COLOR from '../../themes/Color';
import { TrackEventSource } from '../../models/Generics.d';
import { trackEvent } from '../../utils/AppCenterTool';

interface CollapsibleHeaderCardProps {
  title: string;
  isOpened: boolean;
  toggleIsOpened: React.Dispatch<React.SetStateAction<boolean>>;
  icon: string;
  titleStyle: TextStyle | undefined;
  trackEventSource: TrackEventSource | undefined;
}

interface CollapsibleCardProps {
  title: string;
  isOpened?: boolean;
  children: ReactNode | ReactElement;
  icon?: string;
  titleStyle?: TextStyle;
  source?: TrackEventSource;
}

export const CollapsibleHeaderCard = (props: CollapsibleHeaderCardProps): JSX.Element => {
  const {
    icon, title, isOpened, toggleIsOpened, titleStyle, trackEventSource
  } = props;
  const iconName = isOpened ? 'keyboard-arrow-up' : 'keyboard-arrow-down';

  return (
    <View style={styles.menuContainer}>
      <TouchableOpacity
        testID="collapsible-card"
        style={styles.titleContainer}
        onPress={() => {
          toggleIsOpened(!isOpened);
          if (trackEventSource) {
            trackEvent(trackEventSource.screen, {
              action: trackEventSource.action,
              ...trackEventSource.otherInfo
            });
          }
        }}
      >
        <View style={styles.arrowView}>
          <Text style={[titleStyle]}>{title}</Text>
        </View>
        <MaterialIcons name={iconName} size={30} color={COLOR.BLACK} />
      </TouchableOpacity>
    </View>
  );
};

export const CollapsibleCard = (props: CollapsibleCardProps): JSX.Element => {
  const { icon, title, children, isOpened, titleStyle, source } = props;

  const [cardOpen, toggleCardOpen] = useState(isOpened || false);

  return (
    <View>
      <CollapsibleHeaderCard
        title={title}
        isOpened={cardOpen}
        toggleIsOpened={toggleCardOpen}
        icon={icon || ''}
        titleStyle={titleStyle}
        trackEventSource={source}
      />
      {cardOpen && (
      <View>
        {children}
      </View>
      )}
    </View>
  );
};

CollapsibleCard.defaultProps = {
  isOpened: false,
  icon: '',
  titleStyle: {},
  source: undefined
};

export default CollapsibleCard;
