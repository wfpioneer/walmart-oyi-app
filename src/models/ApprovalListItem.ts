export interface ApprovalListItem {
  imageUrl?: string;
  itemName: string;
  itemNbr: number;
  upcNbr: number;
  categoryNbr: number;
  categoryDescription: string;
  subCategoryNbr: number;
  subCategoryDescription: string;
  oldQuantity: number;
  newQuantity: number;
  dollarChange: number;
  initiatedUserId: string;
  initiatedTimestamp: string;
  approvalStatus: approvalStatus;
  approvalRequestSource: approvalRequestSource;
  isChecked?: boolean;
  daysLeft?: number;
}

export interface ApprovalCategory extends ApprovalListItem {
  categoryHeader?: boolean;
}

export enum approvalStatus {
  'Pending' = 'Pending',
  'Approved' = 'Approved',
  'Rejected' = 'Rejected',
  'Expired' = 'Expired'
}
export enum approvalRequestSource {
  'ItemDetails' = 'itemdetails',
  'Audits' = 'audits',

}
