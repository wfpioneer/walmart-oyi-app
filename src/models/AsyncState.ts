export interface AsyncState{
  isWaiting: boolean;
  value: any;
  error: any;
  result: any;
}
