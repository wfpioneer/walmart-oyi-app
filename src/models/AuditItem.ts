export interface ItemPalletInfo {
  palletId: string,
  quantity: number,
  sectionId: number,
  locationName: string,
  mixedPallet: boolean,
  newQty?: number
}
