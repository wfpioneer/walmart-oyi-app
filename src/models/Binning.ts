export interface BinningItem {
  itemNbr: number;
  itemDesc: string;
  price: number;
  upcNbr: string;
  quantity: number;
}
export interface BinningPallet {
  id: number;
  expirationDate: string;
  lastLocation?: string;
  firstItem: BinningItem;
}