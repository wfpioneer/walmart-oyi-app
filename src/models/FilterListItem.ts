interface FilterListItem {
    value: string;
    display: string;
    selected: boolean;
  }

export default FilterListItem;
