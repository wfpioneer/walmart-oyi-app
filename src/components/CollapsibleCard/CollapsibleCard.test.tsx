import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import CollapsibleCard, { CollapsibleHeaderCard } from './CollapsibleCard';
import { TrackEventSource } from '../../models/Generics';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');
jest.mock('../../utils/AppCenterTool.ts', () => ({
  ...jest.requireActual('../../utils/__mocks__/AppCenterTool'),
  trackEvent: jest.fn()
}));

describe('Tests CollapsibleCard Component', () => {
  it('should tests CollapsibleCard when the card is opened', () => {
    const { toJSON } = render(
      <CollapsibleCard
        isOpened={true}
        title="History"
      >{`ABC`}</CollapsibleCard>
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should tests CollapsibleCard when the card is closed', () => {
    const { toJSON } = render(
      <CollapsibleCard
        title="History"
      >{`ABC`}</CollapsibleCard>
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should test children of the card when it is opened', async () => {
    const { findByText } = render(
      <CollapsibleCard
        title="History"
      >{`ABC`}</CollapsibleCard>
    );
    expect(findByText('ABC')).toBeTruthy();
  });
});

describe('Tests CollapsibleHeaderCard Component', () => {
  const mockTrackEventSource: TrackEventSource = {
    action: 'test',
    screen: 'testScreen'
  }
  it('should test CollapsibleHeaderCard onPress', () => {
    const mockToggleOpen = jest.fn();
    const { getByTestId } = render(
      <CollapsibleHeaderCard
        toggleIsOpened={mockToggleOpen}
        isOpened={false}
        title="ABAR-1"
        icon="" 
        titleStyle={undefined} 
        trackEventSource={mockTrackEventSource}
        />
    );
    const collapsibleCard = getByTestId('collapsible-card');
    fireEvent.press(collapsibleCard);
    expect(mockToggleOpen).toHaveBeenCalledTimes(1);
    expect(mockToggleOpen).toHaveBeenCalledWith(true);
  });
  it('should test title of the card', () => {
    const { findByText } = render(
      <CollapsibleHeaderCard
        toggleIsOpened={jest.fn}
        isOpened={false}
        title="ABAR-1"
        icon="" 
        titleStyle={undefined} 
        trackEventSource={mockTrackEventSource}
        />
    );
    expect(findByText('ABAR-1')).toBeTruthy();
  });
});
