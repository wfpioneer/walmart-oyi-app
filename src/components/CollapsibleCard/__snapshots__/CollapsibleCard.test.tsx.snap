// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests CollapsibleCard Component should tests CollapsibleCard when the card is closed 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ffffffff",
        "borderBottomColor": "#eeeeee",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="collapsible-card"
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            [
              {},
            ]
          }
        >
          History
        </Text>
      </View>
      <Icon
        color="#000000"
        name="keyboard-arrow-down"
        size={30}
      />
    </View>
  </View>
</View>
`;

exports[`Tests CollapsibleCard Component should tests CollapsibleCard when the card is opened 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ffffffff",
        "borderBottomColor": "#eeeeee",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="collapsible-card"
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            [
              {},
            ]
          }
        >
          History
        </Text>
      </View>
      <Icon
        color="#000000"
        name="keyboard-arrow-up"
        size={30}
      />
    </View>
  </View>
  <View>
    ABC
  </View>
</View>
`;
